### PR TITLE
Make Travis tests less troublesome.

### PR DESCRIPTION
**Pull request by matejak**
Prevent harmless or expected failiures to result in failed test run.

### CODE REVIEW

**Comment by mpreisler** [comment]
ACK when jenkins finishes

**Comment by matejak** [comment]
And Travis, to be really sure that it works :-)

File: .travis.yml [hunk 1]
```diff
@@ -13,10 +13,10 @@ before_script:
   - cd build
 
 script:
-  - ../configure CFLAGS="--coverage -ftest-coverage -fprofile-arcs" LDFLAGS=-lgcov --enable-debug && build-wrapper-linux-x86-64 --out-dir bw-output make all
+  - ../configure CFLAGS="--coverage -ftest-coverage -fprofile-arcs" LDFLAGS=-lgcov --enable-debug && build-wrapper-linux-x86-64 --out-dir bw-output make all || true  # Will always fail on external PRs.
```
Comment (mpreisler) [inline .travis.yml]: could we split configure step and build step into two lines, makes it much easier to read. we can insist on configure passing and maybe `|| true` only the build part

File: .travis.yml [hunk 2]
```diff
@@ -13,10 +13,10 @@ before_script:
...
   - make -k check VERBOSE=true || { test "$SEND_COVERAGE_EVEN_IF_TESTS_FAIL" = yes && curl -s https://codecov.io/bash > cov.sh && bash cov.sh -x "$GCOV"; }
-  - bash ../tests/recursively_generate_gcov.sh .
-  - cd .. && sonar-scanner
+  - bash ../tests/recursively_generate_gcov.sh . || true  # Failiures that occur here are harmless
```
Comment (mpreisler) [inline .travis.yml]: typo

File: .travis.yml [hunk 3]
```diff
@@ -13,10 +13,10 @@ before_script:
...
+  - cd .. && sonar-scanner || true  # Will always fail on external PRs.
```
Comment (mpreisler) [inline .travis.yml]: since we only have external PRs, does it make sense to even run this? I assume you mean forks.

**Comment by mpreisler** [comment]
Note for later: This will need a merge into master and then another PR in master to change it to use cmake and ctest.

File: .travis.yml (hunk 3 above)
Comment (matejak) [reply .travis.yml]: It makes sense, as merge also triggers a build, which won't fail.

File: .travis.yml (hunk 1 above)
Comment (matejak) [reply .travis.yml]: Good idea!

**Pull request merged**